packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -63,9 +65,22 @@ func GetProtocolStateVariables(
 	opts.Context, cancel = CtxWithTimeoutOrDefault(opts.Context, defaultTimeout)
 	defer cancel()
 
+	var slotBV1 bindings.TaikoDataSlotBV1
 	slotA, slotB, err := taikoL1Client.GetStateVariables(opts)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, ErrSlotBMarshal) {
+			slotA, slotBV1, err = taikoL1Client.GetStateVariablesV1(opts)
+			slotB = bindings.TaikoDataSlotB{
+				NumBlocks:           slotBV1.NumBlocks,
+				LastVerifiedBlockId: slotBV1.LastVerifiedBlockId,
+				ProvingPaused:       slotBV1.ProvingPaused,
+				LastProposedIn:      nil,
+				LastUnpausedAt:      slotBV1.LastUnpausedAt,
+			}
+		}
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Seems there are still some places are using the raw `c.TaikoL1.GetStateVariables`, lets also change them to use the method here?

Reply:
Yes, but we also see the raw `GetStateVariables ` used in Indexer.
Maybe we should make Jeff noticed?

Reply:
> Yes, but we also see the raw `GetStateVariables ` used in Indexer. Maybe we should make Jeff noticed?

Correct, cc @cyberhorsey